### PR TITLE
Add official support for python 3.13

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Adds support for python 3.13. 

**NOTE:** This PR should be merged after https://github.com/kedro-org/kedro-viz/pull/2190

## Development notes

* Update workflows to include python 3.13 build
* Update release note
* Update version support warning

## QA notes

* All tests should pass

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ Please follow the established format:
 
 ## Major features and improvements
 
+- Add support for Python 3.13
+- Add support for Python 3.12
```
Comment (rashidakanchwala) [inline RELEASE.md:14]: maybe after merge we put this in one line with both PR numbers seperated by commas

File: RELEASE.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Please follow the established format:
 
 ## Major features and improvements
 
-- Add support for Python 3.12
+- Add support for Python 3.12, 3.13
```
Comment (rashidakanchwala) [inline RELEASE.md:13]: in the brackets, can you add the PRs

**Review by rashidakanchwala — APPROVED** [review]
Looks good to me!

**Review by SajidAlamQB — APPROVED** [review]
Thank you @ravi-kumar-pilla, awesome to see 3.13 support!

**Pull request merged**